lib/krane/deploy_task.rb

```diff
@@ -277,16 +277,42 @@ def validate_configuration(prune:)
     end
     measure_method(:validate_configuration)
 
+    def partition_dry_run_resources(resources)
+      individuals = []
+      mutating_webhooks = cluster_resource_discoverer.fetch_mutating_webhook_configurations
+      mutating_webhooks.each do |spec|
+        spec.dig('webhooks').each do |webhook|
+          match_policy = webhook.dig('matchPolicy')
+          webhook.dig('rules').each do |rule|
+            next if %w(None NoneOnDryRun).include?(rule.dig('sideEffects'))
+            groups = rule.dig('apiGroups')
+            versions = rule.dig('apiVersions')
+            kinds = rule.dig('resources').map(&:singularize)
+            groups.each do |group|
+              versions.each do |version|
+                kinds.each do |kind|
+                  individuals += resources.select do |r|
+                    (r.group == group || group == '*' || match_policy == "Equivalent") &&
+                    (r.version == version || version == '*' || match_policy == "Equivalent") &&
+                    (r.type.downcase == kind.downcase)
+                  end
```

Comment:
This is a straightforward implementation of the rules defined [here](https://kubernetes.io/docs/reference/kubernetes-api/extend-resources/mutating-webhook-configuration-v1/#MutatingWebhookConfiguration).

Reply:
Any idea how to ask the api server what GVKs it knows to convert between? Not saying we use them, but this adds to our list of things that break with non-unique kinds.

Reply:
>Any idea how to ask the api server what GVKs it knows to convert between?

From my understanding, the `Equivalent` condition simply disregards gvk (I'm tempted to open an upstream issue since this can have undesirable side-effects as we've found). I don't think this is something for Krane to deal with.

From the official reference:
>Equivalent: match a request if modifies a resource listed in rules, even via another API group or version. For example, if deployments can be modified via apps/v1, apps/v1beta1, and extensions/v1beta1, and "rules" only included apiGroups:["apps"], apiVersions:["v1"], resources: ["deployments"], a request to apps/v1beta1 or extensions/v1beta1 would be converted to apps/v1 and sent to the webhook.